packages/nextjs-template/utils/build.ts

```diff
@@ -32,48 +36,65 @@ export function getNoteBody(id: string) {
 
 let _NOTES_CACHE: NoteData | undefined;
 
-export function getNotes() {
+export function getNotes(): NoteData {
   if (_.isUndefined(_NOTES_CACHE)) {
     const dataDir = getDataDir();
     _NOTES_CACHE = fs.readJSONSync(
-      path.join(dataDir, "notes.json")
+      path.join(dataDir, "notes.json"),
     ) as NoteData;
   }
   return _NOTES_CACHE;
 }
 
+export interface DendronNotePageParams extends ParsedUrlQuery {
+  id: string;
+}
 /**
  * Generate URLs for all exported pages
+ * For use with getStaticProps
+ * https://nextjs.org/docs/basic-features/data-fetching/get-static-props
  * @returns
  */
-export function getNotePaths() {
+export function getNotePaths(): GetStaticPathsResult<DendronNotePageParams> {
   const { notes, noteIndex } = getNotes();
-  const ids = _.reject(_.keys(notes), (id) => id === noteIndex.id);
+  // filter out the index node
+  const paths = Object.keys(notes)
+    .filter((id) => id !== noteIndex.id)
+    .map(
+      (id) => {
+        return { params: { id } };
+      },
+    );
   return {
-    paths: _.map(ids, (id) => {
-      return { params: { id } };
-    }),
+    paths,
     fallback: false,
   };
 }
 
-export function getNoteMeta(id: string) {
+/**
+ * Reads the JSON contents of data/meta/<note>.json
+ */
+export function getNoteMeta(id: string): Promise<NoteProps> {
   const dataDir = getDataDir();
   return fs.readJSON(
-    path.join(dataDir, NOTE_META_DIR, `${id}.json`)
-  ) as Promise<NoteProps>;
+    path.join(dataDir, NOTE_META_DIR, `${id}.json`),
+  );
 }
 
+let _CONFIG_CACHE: IntermediateDendronConfig | undefined;
```

Comment:
getConfig() is called a lot during the build process... multiple times for each note